### DISCUSSION TITLE
Should we create a custom configuration file to pass in configuration via environment variables?

### DISCUSSION THREAD

**Discussion by davidbarratt** [phabricator-docker/phabricator#47] [Ideas]
I noticed that Phabricator allows [custom configuration files](https://secure.phabricator.com/book/phabricator/article/advanced_configuration/) to be created. We could create a custom config file that allows __some__ configuration to be passed in through environment variables. The user could always override this file completely (or the configuration it imposes).

Is this something we should do? Would it be helpful? If so, should we use a prefix for the environment variables like `PHABRICATOR_` ?

**Comment by d0niek** [comment]
But do you want to create custom config file from env variables every time when container start (entrypoint) or only once for fresh installation (entrypoint + `if ($isFreshInstall)`)?

**Reply by davidbarratt** [reply]
I was thinking we'd make like a `docker.conf.php` that is always in the container and set with the `PHABRICATOR_ENV` environment variable. The local/database config always takes precidence anyways, or you can set `PHABRICATOR_ENV` to an empty string.

**Comment by d0niek** [comment]
I see that Wordpress docker image from version 5.7 (?) has file [wp-config-docker.php](https://github.com/docker-library/wordpress/blob/394c79eafd5b345514f87cec590577e641c030ef/wp-config-docker.php) and use it as **wp-config.php**. So they read env variables every time when container start

**Reply by davidbarratt** [reply]
Yeah that's interesting, they write the file on startup, I was thinking we'd just read the environemtn variables directly with `$_SERVER`

**Reply by d0niek** [reply]
Hm.. if you want to read env variables from `$_SERVER` you have to setup them in apache?
Is not better to read env variables with `getenv()`? With this we can setup env in docker-compose for php.

**Reply by davidbarratt** [reply]
oh right, yeah that's probably a better option, but I think we'd have to add a `PassEnv` directive for apache.

**Reply by d0niek** [reply]
Hm.. as I understood, env `PHABRICATOR_ENV` have to setup in webserver (apache) so yes.
